mreg/api/v1/tests/test_host_permissions.py

```diff
@@ -1,21 +1,41 @@
+from unittest import skip
+
 from django.contrib.auth.models import Group
+
 from rest_framework import exceptions
 
-from mreg.api.permissions import get_settings_groups
 from mreg.models.host import Host, Ipaddress, PtrOverride
+from mreg.models.auth import MregAdminGroup
 from mreg.models.network import NetGroupRegexPermission, Network
 from mreg.models.zone import ForwardZone
 
 from .tests import MregAPITestCase
 
 
-class Internals(MregAPITestCase):
+# None of these tests work as overriding settings is still weirdly broken:
+# https://docs.djangoproject.com/en/5.1/topics/testing/tools/#overriding-settings
+# We still get the default value of the SUPERUSER groups: 'default-super-group'
+class Internals(MregAPITestCase):    
     """Test internal structures in permissions."""
+    
+    @skip("Overriding settings is broken")
     def test_missing_group_settings(self):
         """Ensure that missing group settings are caught if requested."""
         with self.assertRaises(exceptions.APIException):
-            get_settings_groups("NO_SUCH_SETTINGS_GROUP")
-
+            self.settings(SUPERUSER_GROUP=None)
+            MregAdminGroup.SUPERUSER.settings_groups_or_raise()
+
+    @skip("Overriding settings is broken")
+    def test_groups_always_a_list_from_str(self):
+        """Ensure that the settings are always a list, even if there is only one group."""
+        self.settings(SUPERUSER_GROUP='superuser')
+        self.assertEqual(MregAdminGroup.SUPERUSER.settings_groups_or_raise(), ['superuser'])
+
+    @skip("Overriding settings is broken")
+    def test_groups_always_a_list_from_list(self):
+        """Ensure that the settings are always a list, even if there is only one group."""
+        self.settings(SUPERUSER_GROUP=['superuser'])
```

Comment:
This should work if used as a context provider, i.e.
```
        with self.settings(SUPERUSER_GROUP=['superuser']):
            self.assertEqual(MregAdminGroup.SUPERUSER.settings_groups_or_raise(), ['superuser'])
```
The alternative is to decorate the class or methods:
```
@override_settings(SUPERUSER_GROUP=['superuser'])
...
```

Reply:
Yay, thanks! The decorators worked fine!